DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/IncidentProducer.cs

```diff
@@ -0,0 +1,215 @@
+                const string container = "extra-parameters";
+                string Blob = incidentID + "\\" + param; // ID unique to the incident
+                var blobStorageConnectionString = Environment.GetEnvironmentVariable("BlobStorageConnectionString");
+                var blobStorageKeys = Environment.GetEnvironmentVariable("BlobStorageAccountKeys");
+                WriteData(blobStorageConnectionString, container, Blob, value);
+            }
+            catch (Exception ex)
+            {
+                log.LogError("Exception --> " + ex.Message);
+            }
+        }
+
+        [FunctionName("IncidentProducer")]
+        public static async Task RunAsync(
+#if DEBUG
+            [TimerTrigger("*/30 * * * * *")] TimerInfo myTimer,
+#else
+            [TimerTrigger("* */5 * * * *")]TimerInfo myTimer,
+#endif
+            [Queue("%CohesityQueueName%"), StorageAccount("AzureWebJobsStorage")] ICollector<string> outputQueueItem,
+            ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+            long startDateUsecs = 0;
+
+            try
+            {
+                var db = Connection.GetDatabase();
+                string apiKey = Environment.GetEnvironmentVariable("apiKey");
+                string redisKey = Environment.GetEnvironmentVariable("workspace") + apiKey;
+
+                try
+                {
+                    startDateUsecs = long.Parse(db.StringGet(redisKey));
+                }
+                catch (Exception ex)
+                {
+                    startDateUsecs = GetPreviousUnixTime(log);
+                    log.LogError("Exception --> 2 " + ex.Message);
+                }
+
+                if (startDateUsecs == 0)
+                {
+                    startDateUsecs = GetPreviousUnixTime(log);
+                }
+
+                log.LogInformation("startDateUsecs --> " + startDateUsecs);
+
+                long endDateUsecs = GetCurrentUnixTime();
+                log.LogInformation("endDateUsecs --> " + endDateUsecs.ToString());
+
+                string requestUriString = $"https://helios.cohesity.com/mcm/alerts?alertCategoryList=kSecurity&alertStateList=kOpen&startDateUsecs={startDateUsecs}&endDateUsecs={endDateUsecs}";
+                log.LogInformation("requestUriString --> " + requestUriString);
+                using HttpClient client = new ();
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Add("apiKey", System.Environment.GetEnvironmentVariable("apiKey"));
+                await using Stream stream = await client.GetStreamAsync(requestUriString);
+                StreamReader reader = new StreamReader(stream);
+                dynamic alerts = JsonConvert.DeserializeObject(reader.ReadToEnd());
+
+                foreach (var alert in alerts)
+                {
+                    await ParseAlertToQueueAsync(outputQueueItem, alert, log);
```

Comment:
Why are you doing it sequentially now? There was a code here that did all iterations in parallel.